jujugui/static/gui/src/app/components/inspector-change-version/test-inspector-change-version.js

```diff
@@ -0,0 +1,202 @@
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+var juju = {components: {}};
+var testUtils = React.addons.TestUtils;
+
+chai.config.includeStack = true;
+chai.config.truncateThreshold = 0;
+
+describe('InspectorChangeVersion', function() {
+
+  beforeAll(function(done) {
+    // By loading this file it adds the component to the juju components.
+    YUI().use('inspector-change-version', function() { done(); });
+  });
+
+  it('can display an empty versions list', function() {
+    var changeState = sinon.stub();
+    var service = sinon.stub();
+    var setCharm = sinon.stub();
+    var getCharm = sinon.stub();
+    var getAvailableVersions = sinon.stub().callsArgWith(1, ['cs:django']);
+    var shallowRenderer = jsTestUtils.shallowRender(
+        <juju.components.InspectorChangeVersion
+          changeState={changeState}
+          charmId="cs:django"
+          service={service}
+          setCharm={setCharm}
+          getCharm={getCharm}
+          getAvailableVersions={getAvailableVersions} />, true);
+    shallowRenderer.getMountedInstance().componentDidMount();
+    var output = shallowRenderer.getRenderOutput();
+    assert.deepEqual(output,
+      <div className="inspector-change-version">
+        <div className="inspector-change-version__current">
+          Current version:
+          <div className="inspector-change-version__current-version"
+            role="button" tabIndex="0"
+            onClick={output.props.children[0].props.children[1].props.onClick}>
+            cs:django
+          </div>
+        </div>
+        <ul className="inspector-change-version__versions">
+          <li className="inspector-change-version__none">
+            No other versions.
+          </li>
+        </ul>
+      </div>);
+  });
+
+  it('can display list of versions', function() {
+    var changeState = sinon.stub();
+    var service = sinon.stub();
+    var setCharm = sinon.stub();
+    var getCharm = sinon.stub();
+    var getAvailableVersions = sinon.stub().callsArgWith(1, [
+      'cs:django-4', 'cs:django-5', 'cs:django-6'
+    ]);
+    var shallowRenderer = jsTestUtils.shallowRender(
+        <juju.components.InspectorChangeVersion
+          changeState={changeState}
+          charmId="cs:django-5"
+          service={service}
+          setCharm={setCharm}
+          getCharm={getCharm}
+          getAvailableVersions={getAvailableVersions} />, true);
+    shallowRenderer.getMountedInstance().componentDidMount();
+    var output = shallowRenderer.getRenderOutput();
+    var list = output.props.children[1];
+    assert.deepEqual(output,
+      <div className="inspector-change-version">
+        <div className="inspector-change-version__current">
+          Current version:
+          <div className="inspector-change-version__current-version"
+            role="button" tabIndex="0"
+            onClick={output.props.children[0].props.children[1].props.onClick}>
```

Comment:
Leaving aside the "copying elements of the actual object into the expected object" comment aside, I'd also like to see longer bits like this broken down into variables for understandability.

~~~~
var changeVersion = output.props.children[0];
var currentVersion = changeVersion.props.children[1];
...
onClick={currentVersion.props.onClick}
~~~~